pkg/agent/puller.go

```diff
@@ -179,27 +184,35 @@ func (p *Puller) modelProcessor(modelName string, ops <-chan *ModelOp) {
 			// If there is an error, we will NOT do a delete... that could be problematic
 			if err := storage.RemoveDir(filepath.Join(p.Downloader.ModelDir, modelName)); err != nil {
 				p.logger.Error(err, "failing to delete model directory")
+				break
+			}
+			// unload model from model server
+			resp, err := http.Post(fmt.Sprintf("http://localhost:8080/v2/repository/models/%s/unload", modelName),
+				"application/json",
+				bytes.NewBufferString("{}"))
+			if err != nil {
+				// handle error
+				p.logger.Errorf("Failed to Unload model %s", modelName)
 			} else {
-				// unload model from model server
-				resp, err := http.Post(fmt.Sprintf("http://localhost:8080/v2/repository/models/%s/unload", modelName),
-					"application/json",
-					bytes.NewBufferString("{}"))
-				if err != nil {
-					// handle error
-					p.logger.Errorf("Failed to Unload model %s", modelName)
+				defer func(Body io.ReadCloser) {
+					newErr := Body.Close()
+					if newErr != nil {
+						p.logger.Error(newErr, "failed to close body")
+					}
+				}(resp.Body)
```

Comment:
if resp is nil then this will panic
 ```go
 if resp != nil {
    defer resp.Body.Close()
 }
 ```

Reply:
also is it a good practice to check the error from `Close()`? most of the golfing examples I have seen does not check this

Reply:
Updated. I have seen it quite often in other projects.